### DISCUSSION TITLE
Bit-field TM/TC quantities

### DISCUSSION THREAD

**Discussion by JackNWhite** [nasa/fprime#2318] [Ideas]
Are there any plans to handle bit-fields in TM/TC, in F' and/or FPP? The saving in data can be significant.

**Comment by LeStarch** [comment]
I am not familiar with TM/TC?  Is this another name for channels and events?

One could always define a bit-field, a hex format, and cast to place these items in telemetry.  E.g.:

**Note: pseudo code**
```
union BitField1 {
    U32 field1:8;
    U32 field2:8;
...
}


log_U32_Event(static_cast<U32>(bit_field_var));
```

Does this suffice?

**Reply by JackNWhite** [reply]
Hi, TM/TC is telemetry/telecommanding.

I have a large multi-byte bit-field of which each bit corresponds to the status of a variable which is part of an autocoded struct. The difficulty isn't in downlinking the data, but rather in viewing it in GDS.

It would be great to be able to add a type to FPP like your union. Then I could generate the FPP with my autocoder. The principle would be the same regardless of my circumstances, though.

```
struct MyStatusBitField {
   statusVar1: U1,
   statusVar2: U1,
   statusVar3: U1,
   statusVar4: U1,
   statusVar5: U1,
   statusVar6: U1,
   ...
}
```

I suppose there could also be a way to handle padding, either by just calling that member `paddingX`, or perhaps declaring it `_: UX`.

Do you think it would be difficult to add sub-byte types `U1`..`U7` to FPP and have GDS and F' work out the masks for the different values?
